src/Commands/TankDrive.cpp

```diff
@@ -20,8 +20,30 @@ void TankDrive::Execute() {
   double right = oi->getDriverRightYAxis();
 
   Chassis::getInstance()->SetTankDrive(left, right);
-}
 
+  //Automatic Shifters
+ SmartDashboard::PutNumber("Right Drive Encoder Value",Chassis::getInstance()->GetRightEncRPM());
+ SmartDashboard::PutNumber("Right Throttle Value", right);
+ SmartDashboard::PutNumber("Left Throttle Value", left);
+  double shiftUpVelocity =  1000;
+
+  double shiftDownVelocity = 500;
+
+  if ((Chassis::getInstance()->IsShifterHigh() == !Chassis::ShifterValue::kShifterHigh)
```

Comment:
Let's revisit this logic. Consider checking the shifter state separately from the speed. This should make the tests go much faster under the most usual situation of steady state. 

(I'm also correcting the syntax errors and reversing the logic because it makes more sense.)

```
if (chassis->IsShifterHigh())
{
  if (chassis->GetLeftEncRPM() <= kShiftDownVelocity 
      || chassis->GetRightEncRPM() <= kShiftDownVelocity)
  {
    chassis->SetShifter(Chassis::kShifterLow);
  }
}
else
{
  if (chassis->GetLeftEncRPM() >= kShiftUpVelocity
      || chassis->GetRightEncRPM() >= kShiftUpVelocity)
  {
    chassis->SetShifter(Chassis::kShifterHigh);
  }
}
```

We should go over this code, in order to understand the changes and why I suggest them.